### DISCUSSION TITLE
Improvement for reoccuring payments / recurrent records

### DISCUSSION THREAD

**Discussion by Vertelleus** [emavgl/oinkoin#171] [Ideas]
## Current behavior
<ins>Note</ins>: This is the current behavior as I have seen in the app. Please let me know if I have missed a setting somewhere that would change the view.
1. Recurrent records are given only a time frame interval to repeat.
2. Recurrent records do not show pending future payments.

## Requested behavior
1. Give recurrent records a start date and end date. Default the start date to date created, and default end date to some form of "does not end"
    - The start date allows a person to set the date of the recurrent record so that it will not interfere with past payments.
    - The end date allows the user to set a limit on when they would like for the record to stop being added.
2. Allow visibility of future payments, possibly with a toggle in the settings.
    - If I am paid every 2 weeks and I pay rent at the end of every month, I would like to see the totals of that in every view (week, month, annual), even if the date of the payment is in the future.

## Conclusion
Thank you for taking the time to look at this!

**Comment by emavgl** [comment]
Hi, thank you for taking the time to write these suggestions! Here are my thoughts:

Currently, it is possible to specify a start date, but there is no defined end date, resulting in an open-ended schedule. This open-ended approach is causing challenges when generating future payment records. The main question is: "When should the creation of records for a recurring payment stop?" As a result, records are only generated as they actually occur.

1. Allowing users to define a specific interval could indeed be a helpful feature, as it would aid in planning and potentially resolve the issue mentioned above. However, I appreciate the flexibility of having an open-ended option and the manual control to delete current records as needed. For this reason, I don’t think it is strictly necessary at this point. That said, I do understand that some users might prefer an automatic deletion of recurrent records, so offering this as an option could be beneficial.

2. Yes, this was a requested feature as well. My current approach is to create records only when they fall within the user’s chosen view. For instance, if a user opts for a yearly view, I would generate all the recurrent records for that year. If the yearly view isn’t selected, no records would be generated.

I am still undecided on the best approach for this and open to further insights or suggestions.
